### ISSUE TITLE
Accessibility improvements to the allocation table collapse functionality

### ISSUE THREAD

**Issue by wrightmartin** [ecosyste-ms/funds#103]
From the Bootstrap docs

> Accessibility 
> Be sure to add aria-expanded to the control element. This attribute explicitly conveys the current state of the collapsible element tied to the control to screen readers and similar assistive technologies. If the collapsible element is closed by default, the attribute on the control element should have a value of aria-expanded="false". If you’ve set the collapsible element to be open by default using the show class, set aria-expanded="true" on the control instead. The plugin will automatically toggle this attribute on the control based on whether or not the collapsible element has been opened or closed (via JavaScript, or because the user triggered another control element also tied to the same collapsible element). If the control element’s HTML element is not a button (e.g., an <a> or <div>), the attribute role="button" should be added to the element.
> 
> If your control element is targeting a single collapsible element – i.e. the data-bs-target attribute is pointing to an id selector – you should add the aria-controls attribute to the control element, containing the id of the collapsible element. Modern screen readers and similar assistive technologies make use of this attribute to provide users with additional shortcuts to navigate directly to the collapsible element itself.

In this specific instance I think it's a matter of

- [x] adding `role="button"` to the anchor tag
- [x] add the `aria-controls` attribute to the control element (`<tr class="clickable-row"`) containing the id of the collapsible element (`id="details-<%= i %>"`)
- [x] Add and toggle the `aria-expanded=""`

**Comment by wrightmartin** [comment]
Just realised it already did item #3 so this was a very easy fix
